### ISSUE TITLE
Extend selector-combinator-blacklist to ignore it for pseudo classes

### ISSUE THREAD

**Issue by alexander-schranz** [stylelint/stylelint#4276]
> What is the problem you're trying to solve?

I want to use the following:

```json
"selector-combinator-blacklist": [" "],
```

But want to allow it if the parent has a pseudo-class sector on it like :hover, :focus, ...

Example the following should not violate the rule

```css
.block:hover .block__element {
}
```

The following should:

```css
.block .block__element:hover {
}

.block:hover .block__element .block__other_element {
}
```

> What solution would you like to see?

Any additional configuration that allow ignoring the rule for pseudo-classes?

**Comment by hudochenkov** [comment]
It's a straight way to [write a plugin](https://stylelint.io/developer-guide/plugins), because it's not generally applied and tied to a specific methodology.

Side note: You probably are using BEM. You can't target element inside other element: `.block:hover .block__element .block__other_element`. You can do `.block:hover .block__element` or `.block:hover .block__other_element`.

**Comment by jeddy3** [comment]
> It's a straight way to write a plugin, 

Agreed. It's a great candidate for a plugin and a good example of why the plugin system exists.

**Comment by alexander-schranz** [comment]
@hudochenkov @jeddy3 will give it a try thank you for your really fast response :)
